extensions/mailer/runtime/src/main/java/io/quarkus/mailer/runtime/MailClientProducer.java

```diff
@@ -56,10 +59,71 @@ public void stop() {
     }
 
     private MailClient mailClient(Vertx vertx, MailConfig config, TlsConfig tlsConfig) {
+        System.setProperty("vertx.mail.attachment.cache.file", String.valueOf(config.cacheAttachments));
+
         io.vertx.ext.mail.MailConfig cfg = toVertxMailConfig(config, tlsConfig);
         return MailClient.createShared(vertx, cfg);
     }
 
+    private io.vertx.ext.mail.DKIMSignOptions toVertxDkimSignOptions(DkimSignOptionsConfig optionsConfig) {
+        DKIMSignOptions vertxDkimOptions = new io.vertx.ext.mail.DKIMSignOptions();
+
+        vertxDkimOptions.setSignAlgo(DKIMSignAlgorithm.RSA_SHA256);
+        vertxDkimOptions.setSdid(optionsConfig.sdid);
+        vertxDkimOptions.setSelector(optionsConfig.selector);
+
+        if (optionsConfig.auid.isPresent()) {
+            vertxDkimOptions.setAuid(optionsConfig.auid.get());
+        }
+
+        if (optionsConfig.bodyLimit.isPresent()) {
+            int bodyLimit = optionsConfig.bodyLimit.getAsInt();
+            if (bodyLimit <= 0) {
+                throw new ConfigurationException("Body limit must be greater than zero.");
+            }
+            vertxDkimOptions.setBodyLimit(bodyLimit);
```

Comment:
Just a heads-up that `io.vertx.ext.mail.DKIMSignOptions#setBodyLimit` already performs this validation :)